ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -387,13 +411,17 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
             data={data}
             seriesMapping={seriesMapping}
             wrapLabels={tooltipConfig.wrapLabels}
-            enablePinning={tooltipConfig.enablePinning}
+            enablePinning={isPinningEnabled}
             pinnedPos={tooltipPinnedCoords}
             unit={unit}
             onUnpinClick={() => {
+              // Unpins tooltip when clicking Pin icon in TooltipHeader.
               setTooltipPinnedCoords(null);
-              // Clear previously set pinned crosshair
-              seriesMapping.pop();
+              // Clear previously set pinned crosshair.
+              const isCrosshairPinned = checkCrosshairPinnedStatus(seriesMapping);
```

Comment:
Should this be `updatedSeriesMapping`? Sorry if making a ton of these is annoying! Hoping it'll help with two eyes trying to catch them.